src/deepsparse/pipeline.py

```diff
@@ -63,11 +64,362 @@
+                and pipeline_constructor.default_model_path
+            ):
+                model_path = pipeline_constructor.default_model_path
+
+            if model_path is None:
+                raise ValueError(
+                    f"No model_path provided for pipeline {pipeline_constructor}. Must "
+                    "provide a model path for pipelines that do not have a default "
+                    "defined"
+                )
+
+            kwargs["model_path"] = model_path
+
+        if issubclass(
+            pipeline_constructor, Bucketable
+        ) and pipeline_constructor.should_bucket(**kwargs):
+            if kwargs.get("input_shape", None):
+                raise ValueError(
+                    "Overriding input shapes not supported with Bucketing enabled"
+                )
+            if not kwargs.get("context", None):
+                context = Context(num_cores=kwargs["num_cores"])
+                kwargs["context"] = context
+            buckets = pipeline_constructor.create_pipeline_buckets(
+                task=task,
+                **kwargs,
+            )
+            return BucketingPipeline(pipelines=buckets)
+
+        return pipeline_constructor(**kwargs)
+
+    @classmethod
+    def register(
+        cls,
+        task: str,
+        task_aliases: Optional[List[str]] = None,
+        default_model_path: Optional[str] = None,
+    ):
+        """
+        Pipeline implementer class decorator that registers the pipeline
+        task name and its aliases as valid tasks that can be used to load
+        the pipeline through `BasePipeline.create()` or `Pipeline.create()`
+
+        Multiple pipelines may not have the same task name. An error will
+        be raised if two different pipelines attempt to register the same task name
+
+        :param task: main task name of this pipeline
+        :param task_aliases: list of extra task names that may be used to reference
+            this pipeline. Default is None
+        :param default_model_path: path (ie zoo stub) to use as default for this
+            task if None is provided
+        """
+        task_names = [task]
+        if task_aliases:
+            task_names.extend(task_aliases)
+
+        task_names = [task_name.lower().replace("-", "_") for task_name in task_names]
+
+        def _register_task(task_name, pipeline_class):
+            if task_name in _REGISTERED_PIPELINES and (
+                pipeline_class is not _REGISTERED_PIPELINES[task_name]
+            ):
+                raise RuntimeError(
+                    f"task {task_name} already registered by BasePipeline.register. "
+                    f"attempting to register pipeline: {pipeline_class}, but"
+                    f"pipeline: {_REGISTERED_PIPELINES[task_name]}, already registered"
+                )
+            _REGISTERED_PIPELINES[task_name] = pipeline_class
+
+        def _register_pipeline_tasks_decorator(
+            pipeline_class: Union[BasePipeline, Pipeline]
```

Comment:
since `Pipeline` inherits from `BasePipeline` we shouldn't need to union `Pipeline` in going forward. unless you have an explicit reason here?